pkg/cli/cliexecuter.go

```diff
@@ -1,97 +1,168 @@
 package cli
 
 import (
-	"bytes"
-	"errors"
+	"context"
+	"fmt"
+	"os/exec"
+	"strings"
+	"sync"
+	"time"
 
 	"github.com/go-logr/logr"
 
-	corev1 "k8s.io/api/core/v1"
-	"k8s.io/client-go/kubernetes"
-	"k8s.io/client-go/kubernetes/scheme"
-	restclient "k8s.io/client-go/rest"
-	"k8s.io/client-go/tools/remotecommand"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
 	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/medik8s/fence-agents-remediation/api/v1alpha1"
+	"github.com/medik8s/fence-agents-remediation/pkg/utils"
 )
 
-type Executer interface {
-	Execute(pod *corev1.Pod, command []string) (stdout string, stderr string, err error)
+type Executer struct {
+	client.Client
+	ctx      context.Context
+	log      logr.Logger
+	routines map[types.UID]bool
+	mapLock  sync.Mutex
+	runner   runnerFunc
 }
 
-type executer struct {
-	log       logr.Logger
-	config    *restclient.Config
-	clientSet *kubernetes.Clientset
+type ShellCommand interface {
+	Run() error
 }
```

Comment:
Is it used somewhere or is it redundant?

Reply:
Not needed indeed, thank you for catching it